qiskit/pulse/channels/pulse_channel_spec.py

```diff
@@ -0,0 +1,155 @@
+    This class can be initialized with two methods shown as below.
+
+    1. When you have BaseBackend object of the target quantum device.
+        ```python
+        system = PulseSpecification.from_device(backend)
+        ```
+
+    2. When you know the number of elements constituting the target quantum device.
+        ```python
+        system = PulseSpecification(n_qubits=5, n_control=6, n_registers=1, buffer=10)
+        ```
+
+    Within Qiskit a quantum system at the level of pulses is abstracted as a combination of multiple
+    types of channels on which instructions are scheduled. These most common channel types are the :
+      - `PulseChannel`: For performing stimulus of the system.
+      - `AcquireChannel`: For scheduling acquisition of qubit data.
+      - `MemorySlot`: For persistent storage of measurement results.
+      - `RegisterSlot`: For temporary storage of and conditional feedback on measurement results.
+
+    There are also several special types of pulse channels which are the:
+      - `DriveChannel`: Used to control a single qubit.
+      - `MeasureChannel`: Used to perform measurement stimulus of a single qubit.
+      - `ControlChannel`: Used to control an arbitrary Hamiltonian term on the system.
+      Typically a two-qubit interaction.
+
+    A collection of above channels is automatically assembled within `PulseChannelSpec`.
+
+    For example, the zeroth drive channel may be accessed by
+        ```python
+        system.drives[0]
+        ```
+    or if the channel is connected to the first qubit,
+        ```python
+        system.qubits[0].drive
+        ```
+    In the above example, both commands refer to the same object.
+    """
+    def __init__(self,
+                 n_qubits: int,
+                 n_control: int,
+                 n_registers: int,
+                 buffer: int = 0):
+        """
+        Create pulse specification with number of channels.
+
+        Args:
+            n_qubits: Number of qubits.
+            n_control: Number of control channels.
+            n_registers: Number of classical registers.
+            buffer: Buffer that should be placed between instructions on channel.
+        """
+        self._drives = [DriveChannel(idx, buffer) for idx in range(n_qubits)]
+        self._controls = [ControlChannel(idx, buffer) for idx in range(n_control)]
+        self._measures = [MeasureChannel(idx, buffer) for idx in range(n_qubits)]
+        self._acquires = [AcquireChannel(idx, buffer) for idx in range(n_qubits)]
+        self._mem_slots = [MemorySlot(idx) for idx in range(n_qubits)]
+        self._reg_slots = [RegisterSlot(idx) for idx in range(n_registers)]
+
+        # create mapping information from channels
+        self._topology = SystemTopology(self._drives, self.controls,
+                                        self.measures, self.acquires)
+
+    @classmethod
+    def from_device(cls, backend):
```

Comment:
Do we change this to `from_system` ?

Reply:
Yes, but only in the new pulse channel spec and not the old DeviceSpecification.